### PR TITLE
win: fix include path check

### PR DESCRIPTION
**Pull request by tdusnoki**
We have a drive letter before the first path separator on Windows.

### CODE REVIEW

File: tools/shared/utils/common/utils.cpp
Comment (tdusnoki) [reply tools/shared/utils/common/utils.cpp:442]: If users would write `-IC` then they would still get the error `Include file path must be a full path name` because there is no path separator after the first two characters.

File: tools/shared/utils/common/utils.cpp
Comment (tdusnoki) [reply tools/shared/utils/common/utils.cpp:448]: As I see from my tests it doesn't matter, it works both ways.

File: tools/shared/utils/common/utils.cpp
Comment (isuruf) [reply tools/shared/utils/common/utils.cpp:442]: Sure, but it could still have a path separator because the third character is out of bounds and accessing it undefined behaviour.

**Review by isuruf — APPROVED** [review]

File: tools/shared/utils/common/utils.cpp
Comment (bryanpkc) [inline tools/shared/utils/common/utils.cpp:442]: Did you consider performing a stricter check here, e.g. `!(isalpha(*cp) && *cp == ':' && isPathSeparator(*(cp+2)))`?

Comment (tdusnoki) [reply tools/shared/utils/common/utils.cpp:442]: yes, we could do exactly that, thanks.

Comment (bryanpkc) [reply tools/shared/utils/common/utils.cpp]: You shouldn't have removed the `strlen` check.

**Comment by kiranchandramohan** [comment]
@bryanpkc please merge this once you approve.

**Comment by albertziegenhagel** [comment]
Just wanted to chime in and note that this will not actually detect all possible absolute paths on Windows. This will be missing support for *UNC* paths like `\\machine\mountpoint\file`, which to by best knowledge are mainly used when files on a non-local drive are to be referenced (network drives).

See https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#fully-qualified-vs-relative-paths for some more information.

Please also note that this documentation might give the impression that a path like `\example` is considered *absolute*. This is usually **not true** as far as I know. E.g. both [PathIsRelative](https://docs.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-pathisrelativea) from the Windows API as well as [std::filesystem::path::is_absolute](https://en.cppreference.com/w/cpp/filesystem/path/is_absrel) with Microsofts STL implementation will return `false` for such a path. (To my understanding it is absolute to the current drive, but not *fully qualified* since the drive is not specified.)

Not sure whether you want to apply the necessary changes in this PR or whether you prefer to merge the current, partial support and open an issue for the remaining UNC path detection.

**Review by bryanpkc — APPROVED** [review]

**Comment by bryanpkc** [comment]
Let's merge this now and track the UNC path support in a new issue.

**Pull request merged**
